addons/web_editor/static/lib/odoo-editor/test/spec/editor.test.js

```diff
@@ -2656,6 +2656,24 @@ X[]
                 contentAfter: `<p>a&nbsp;[]</p>`,
             });
         });
+        it('should transform the separate space node preceded by a text node to &nbsp;', async () => {
+            await testEditor(BasicEditor, {
+                contentBefore: `<p id="node-test-container">ab cd</p>`,
+                stepFunction: async editor => {
+                    const testContainer = editor.document.querySelector("#node-test-container");
+                    splitTextNode(testContainer.firstChild, 2); // "ab"" cd"
+                    const sel = editor.document.getSelection();
+                    const range = editor.document.createRange();
+                    range.setStart(testContainer.lastChild, 1);
+                    range.setEnd(testContainer.lastChild, 3);
+                    sel.removeAllRanges();
+                    sel.addRange(range);
+                    getDeepRange(editor.editable, {sel, splitText: true, select: true, correctTripleClick: true}); // "ab"" ""cd"
+                    await deleteBackward(editor);
+                },
+                contentAfter: `<p id="node-test-container">ab&nbsp;[]</p>`,
```

Comment:
I just ran it locally in 15.0 to check and it indeed doesn't pass before your fix :+1: However, my version passes fine before the fix. Very strange. @Goaman @sebgeelen does this ring any bell?

Reply:
There is no wonder why your test passes. The test you mentioned is not exactly reproducing the bug at all. In website, there are two different text node like this `"ab"" cd"` . This gets transforms into this, `"ab"" ""cd"` by `getDeepRange` because `getDeepRange` is called before `deleteRange`. In order to create exactly same scenario I created the test case like that. Indeed it is a weird case !!

Reply:
@Zynton A gentle reminder for this one, thanks!

Reply:
Sorry I hadn't seen your answer! Thanks for explaining, that makes perfect sense.
Looks good to me, great work! When runbot is green you can mark this task as ready so I can r+.

Reply:
Okay, thanks!